source/common/http/http2/metadata_encoder.h

```diff
@@ -0,0 +1,98 @@
+#pragma once
+
+#include <cstdint>
+#include <string>
+
+#include "common/buffer/buffer_impl.h"
+#include "common/common/logger.h"
+#include "common/http/http2/metadata_interface.h"
+
+namespace Envoy {
+namespace Http {
+namespace Http2 {
+
+/**
+ * A class that creates and sends METADATA payload. The METADATA payload is a group of string key
+ * value pairs encoded in HTTP/2 header block.
+ */
+class MetadataEncoder : Logger::Loggable<Logger::Id::metadata> {
+public:
+  MetadataEncoder(uint64_t stream_id) : stream_id_(stream_id) {}
+
+  /**
+   * Creates wired format HTTP/2 header block from |metadata_map|. Only after previous payload is
```

Comment:
wired format -> wire format?
Also I don't think envoy uses the |variable| syntax - just having raw variable is fine here and below.

Reply:
Fixed.